app/controllers/buildings_controller.ts

```diff
@@ -0,0 +1,87 @@
+import Building from '#models/building'
+import Campus from '#models/campus'
+import { HttpContext } from '@adonisjs/core/http'
+export default class BuildingsController {
+  async getAll({ response }: HttpContext) {
```

Comment:
the function names are okay, but i'd change them to more adonis conventional
if you use `node ace make:controller [name] --api`, you will get the api template

the conventional names are
index - list of all records
store - create new record
show - get individual record
update - quite obvious :p
destroy - deletes record

what i'd suggest - create separate 2 separate controllers to avoid naming conflicts, and change function names to index and store (getByCampus is fine imo)

Reply:
gotcha, will do